api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1155,6 +1167,117 @@ public Appointment bookAppointment(Appointment appointment,
 				appointment);
 	}
 
+
+	//appointment resource
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources() {
+		return getAppointmentResourceDAO().getAll();
+	}
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources(boolean includeVoided) {
+		return getAppointmentResourceDAO().getAllData(includeVoided);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResource(Integer appointmentResourceId) {
+		return (AppointmentResource) getAppointmentResourceDAO().getById(appointmentResourceId);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResourceByUuid(String uuid) {
+		return (AppointmentResource) getAppointmentResourceDAO().getByUuid(uuid);
+	}
+
+	@Override
+	public AppointmentResource saveAppointmentResource(AppointmentResource appointmentResource) throws APIException {
+		ValidateUtil.validate(appointmentResource);
+		return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
+	}
+
+	@Override
+	public AppointmentResource voidAppointmentResource(AppointmentResource appointmentResource, String reason) {
+		appointmentResource.setVoided(true);
+		appointmentResource.setVoidReason(reason);
+		return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
+	}
+
+	@Override
+	public void purgeAppointmentResource(AppointmentResource appointmentResource) {
+		getAppointmentResourceDAO().delete(appointmentResource);
+	}
+
+	@Override
+	public List<AppointmentResource> getAppointmentResourcesByConstraints(Location location, Provider provider, List<AppointmentType> appointmentTypes) {
+		return getAppointmentResourceDAO().getResourceByConstraints(location, provider,null);
+	}
+
+	@Override
+	public TimeSlot getTimeslotForAppointment(Location location, Provider provider, AppointmentType type, Date appointmentDate){
+		TimeSlot requiredTimeslot = null;
+
+		Date dayBeforeAppointmentDate = getDate(appointmentDate, -1);
+		Date dayAfterAppointmentDate = getDate(appointmentDate, 1);
+
+		List<TimeSlot> timeSlots = getTimeSlotsByConstraintsIncludingFull(type, dayBeforeAppointmentDate, dayAfterAppointmentDate, provider, location);
```

Comment:
Wouldn't you just want to all getTimeSlotsByConstraints here, not "getTimeSlotsByConstraintsIncludingFull"?  "Including Full" returns time slots that are full... ie the provider is already fully booked during that time slot.

Reply:
I'm using this getTimeSlotsByConstraintsIncludingFull so that I can display to the user when a suitable timeslot is full, getTimeSlotsByConstraints with not give me that flexibility.

Reply:
That makes sense... but in the implementation in AppointmentDTOController aren't you going ahead and creating the appointment?  Where is the step that you return the time slots to present to the user?

Reply:
its not bout returning the timeslot, its about letting the user know that he can't schedule the appointment in the particular time he has selected because the timeslot there is full

Reply:
so for this to happen i'll have to use getTimeSlotsByConstraintsIncludingFull and also TimeslotFullException  so that i can let the user know that he can't schedule the appointment in the particular time he has selected because the timeslot there is full